Tests/SwiftStdlibTests/CodableExtensionsTests.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  CodableExtensionsTests.swift
+//  SwifterSwift
+//
+//  Created by Mustafa GUNES on 16.01.2020.
+//  Copyright © 2020 SwifterSwift
+//
+
+import XCTest
+
+@testable import SwifterSwift
+
+private struct City: Codable {
+    var cityId: Int
+    var name: String
+    var photoUrl: String
+
+    private enum CodingKeys: String, CodingKey {
+        case cityId = "id"
+        case name
+        case photoUrl = "photo_url"
+    }
+}
+
+final class CodableExtensionsTests: XCTestCase {
+
```

Comment:
Consider adding a test case for invalid data as well, an example: a city with an invalid URL 😉

Reply:
@omaralbeik Thanks :)

Reply:
@omaralbeik  Should Method be wrong?

example:
```
func testDecodeModelInvalidData() {
    guard let city = City.init(from: mockJsonData) else {
        XCTAssert(false, "Could not parse model")
        return
    }

    let invalidURL = "https://cdn.pixabay.com/photo/2017/09/27/20/55/sanliurfa-2793424_1280"

    XCTAssertEqual(city.url, invalidURL)
}
```

Reply:
No, just create an invalid data mock:
```swift
private var invalidMockJsonData: Data {
  return #"{"id": 1, "name": "Istanbul", "url": "invalid url"}"#.data(using: .utf8)!
}
```
and make sure that the initializer fails to create a `City` since the URL is invalid

```swift
XCTAssertNil(City.init(from: invalidMockJsonData))
```

Reply:
@omaralbeik I need to verify the URL for this. The task of the plug-in is to parse the JSON data. Why should the model return empty because of the invalid URL?

Reply:
Sorry my bad, please make the url property of type `URL` instead of `String`